CMakeLists.txt

```diff
@@ -1,29 +1,39 @@
 cmake_minimum_required(VERSION 3.6.3)
+
+# Include custom utilites.
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake;${CMAKE_MODULE_PATH}")
 include(Utilities)
+
 project(KinesisVideoProducerC VERSION 1.5.1 LANGUAGES C)
 
 set(KINESIS_VIDEO_PRODUCER_C_MAJOR_VERSION 1)
 set(KINESIS_VIDEO_PRODUCER_C_MINOR_VERSION 5)
 set(KINESIS_VIDEO_PRODUCER_C_PATCH_VERSION 1)
 set(KINESIS_VIDEO_PRODUCER_C_VERSION ${KINESIS_VIDEO_PRODUCER_C_MAJOR_VERSION}.${KINESIS_VIDEO_PRODUCER_C_MINOR_VERSION}.${KINESIS_VIDEO_PRODUCER_C_PATCH_VERSION})
 
-include(GNUInstallDirs)
+set(SUPPORTED_CRYPTO_LIBRARIES "{OpenSSL, MbedTLS}")
 
+# Set CMAKE_INSTALL_ directories based on GNU standard.
+include(GNUInstallDirs)
 
 # User Flags
 option(ADD_MUCLIBC "Add -muclibc c flag" OFF)
-option(BUILD_STATIC "Static build" OFF)
-option(BUILD_DEPENDENCIES "Whether or not to build depending libraries from source" ON)
 option(USE_OPENSSL "Use openssl as crypto library" ON)
 option(USE_MBEDTLS "Use mbedtls as crypto library" OFF)
-option(BUILD_COMMON_LWS "Whether or not to build ProducerC libwebsockets common library" OFF)
-option(BUILD_COMMON_CURL "Whether or not to build ProducerC curl common library" ON)
+option(BUILD_STATIC "Static build" OFF)
+option(BUILD_DEPENDENCIES "Whether to build dependency libraries from source" ON) # This is a master OFF switch, none will be built if off, but if it is ON, some can be specified to be not built
+option(BUILD_COMMON_LWS "Whether to build ProducerC libwebsockets common library" OFF)
```

Comment:
Nice. For future enhancements, let us see if we can mask these behind a `BUILD_ONLY` overarching flag. Similar to how AWS SDK does: https://github.com/aws/aws-sdk-cpp/blob/main/docs/CMake_Parameters.md#build_only